build/configs/imxrt1020-evk/Make.defs

```diff
@@ -97,31 +96,40 @@ NM = $(CROSSDEV)nm
 OBJCOPY = $(CROSSDEV)objcopy
 OBJDUMP = $(CROSSDEV)objdump
 STRIP = $(CROSSDEV)strip
-
+ifeq ($(CONFIG_BUILD_PROTECTED),y)
+STRIP += --strip-unneeded
+endif
 ARCHCCVERSION = ${shell $(CC) -v 2>&1 | sed -n '/^gcc version/p' | sed -e 's/^gcc version \([0-9\.]\)/\1/g' -e 's/[-\ ].*//g' -e '1q'}
 ARCHCCMAJOR = ${shell echo $(ARCHCCVERSION) | cut -d'.' -f1}
 ARCHCCMINOR = ${shell echo $(ARCHCCVERSION) | cut -d'.' -f2}
 
 ifeq ($(CONFIG_DEBUG_SYMBOLS),y)
   ARCHOPTIMIZATION = -g
 endif
+ifneq ($(CONFIG_DEBUG_NOOPT),y)
+  ARCHOPTIMIZATION += $(MAXOPTIMIZATION)
+endif
 
+ifeq ($(CONFIC_BUILD_PROTECTED),y)
 ifneq ($(CONFIG_DEBUG_NOOPT),y)
-  ARCHOPTIMIZATION += $(MAXOPTIMIZATION) -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer
+  ARCHOPTIMIZATION += -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer
 endif
 
 ifeq ($(CONFIG_FRAME_POINTER),y)
   ARCHOPTIMIZATION += -fno-omit-frame-pointer -mapcs -mno-sched-prolog
 endif
+endif
 
 ARCHCFLAGS = -fno-builtin -mcpu=cortex-m7 -fno-common $(ARCHCPUFLAGS)
-ARCHCXXFLAGS = -fno-builtin -mcpu=cortex-m7 $(ARCHCPUFLAGS)
-ifeq ($(QUICKBUILD),y)
-ARCHWARNINGS = -Wall -Werror -Wstrict-prototypes -Wshadow -Wundef -Wno-implicit-function-declaration -Wno-unused-function -Wno-unused-but-set-variable
-ARCHWARNINGSXX = -Wall -Werror -Wshadow -Wundef
+ARCHCXXFLAGS = -fno-builtin $(ARCHCPUFLAGS)
+ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
+ARCHCXXFLAGS += -fexceptions -mcpu=cortex-m7
 else
-ARCHWARNINGS = -Wall -Wstrict-prototypes -Wshadow -Wundef -Wno-implicit-function-declaration -Wno-unused-function -Wno-unused-but-set-variable
-ARCHWARNINGSXX = -Wall -Werror -Wshadow -Wundef
+ARCHCXXFLAGS += -mcpu=cortex-m7
+endif
```

Comment:
`-mcpu=cortex-m7` is not related with `CONFIG_APP_BINARY_SEPARATION` and it's defined in both. Let's move it out.
```
ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
ARCHCXXFLAGS += -fexceptions
endif
ARCHCXXFLAGS += -mcpu=cortex-m7
```